src/Development/IDE/Session.hs

```diff
@@ -0,0 +1,651 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE CPP #-}
+
+{-|
+The logic for setting up a ghcide session by tapping into hie-bios.
+-}
+module Development.IDE.Session
+#ifndef GHC_LIB
+(loadSession)
+#endif
+where
+
+-- Unfortunately, we cannot use loadSession with ghc-lib since hie-bios uses
+-- the real GHC library and the types are incompatible. Furthermore, when
+-- building with ghc-lib we need to make this Haskell agnostic, so no hie-bios!
+#ifndef GHC_LIB
```

Comment:
@jneira I needed to put back the CPP because cabal seems to compile modules even if they're not listed in the exposed-modules

Reply:
oh, it seems cabal is compiling modules under the common src hs-source-dirs. It bited me when implementing the hlint plugin.
A workaround could be use a diferent hs-source-dirs for Session, and put it under the cabal condition.

Reply:
@bubba have you had a chance to try the previous trick to get ride of cpp in the module?

Reply:
Whoops, forgot to do that, I've put them in the top-level folder `session-loader`